lighthouse-core/audits/screenshot-thumbnails.js

```diff
@@ -67,62 +66,70 @@ class ScreenshotThumbnails extends Audit {
    * @param {!Artifacts} artifacts
    * @return {!AuditResult}
    */
-  static audit(artifacts, context) {
+  static async audit(artifacts, context) {
     const trace = artifacts.traces[Audit.DEFAULT_PASS];
     const cachedThumbnails = new Map();
 
-    return Promise.all([
-      artifacts.requestSpeedline(trace),
-      TTFI.audit(artifacts, context).catch(() => ({rawValue: 0})),
-      TTCI.audit(artifacts, context).catch(() => ({rawValue: 0})),
-    ]).then(([speedline, ttfi, ttci]) => {
-      const thumbnails = [];
-      const analyzedFrames = speedline.frames.filter(frame => !frame.isProgressInterpolated());
-      const maxFrameTime =
-        speedline.complete ||
-        Math.max(...speedline.frames.map(frame => frame.getTimeStamp() - speedline.beginning));
-      // Find thumbnails to cover the full range of the trace (max of last visual change and time
-      // to interactive).
-      const timelineEnd = Math.max(maxFrameTime, ttfi.rawValue, ttci.rawValue);
-
-      for (let i = 1; i <= NUMBER_OF_THUMBNAILS; i++) {
-        const targetTimestamp = speedline.beginning + timelineEnd * i / NUMBER_OF_THUMBNAILS;
-
-        let frameForTimestamp = null;
-        if (i === NUMBER_OF_THUMBNAILS) {
-          frameForTimestamp = analyzedFrames[analyzedFrames.length - 1];
-        } else {
-          analyzedFrames.forEach(frame => {
-            if (frame.getTimeStamp() <= targetTimestamp) {
-              frameForTimestamp = frame;
-            }
-          });
-        }
-
-        const imageData = frameForTimestamp.getParsedImage();
-        const thumbnailImageData = ScreenshotThumbnails.scaleImageToThumbnail(imageData);
-        const base64Data =
-          cachedThumbnails.get(frameForTimestamp) ||
-          jpeg.encode(thumbnailImageData, 90).data.toString('base64');
-
-        cachedThumbnails.set(frameForTimestamp, base64Data);
-        thumbnails.push({
-          timing: Math.round(targetTimestamp - speedline.beginning),
-          timestamp: targetTimestamp * 1000,
-          data: base64Data,
+    const speedline = await artifacts.requestSpeedline(trace);
+
+    let alternativeMaximumValue = 0;
```

Comment:
add a comment why there needs to be an `alternativeMaximumValue`? (there's a comment below but maybe some variant should be up here so you don't spend too long thinking about it :)

Reply:
yeah good call, done